scripts/genvimvim.lua

```diff
@@ -112,10 +112,19 @@ for au, _ in pairs(auevents.aliases) do
   end
 end
 
+local function sorted_keys(tab)
+    local result = {}
+    for key, _ in pairs(tab) do
+        table.insert(result, key)
+    end
+    table.sort(result)
+    return result
+end
+
 local nvimau_start = 'syn keyword nvimAutoEvent contained '
 w('\n\n' .. nvimau_start)
 
-for au, _ in pairs(auevents.nvim_specific) do
+for _, au in ipairs(sorted_keys(auevents.nvim_specific)) do
```

Comment:
similar to what's discussed in https://github.com/neovim/neovim/pull/20355#discussion_r980511837 , can the root cause be fixed instead of sorting at the "leaf logic" here?

Similar question for LUA_GEN_PRG

Reply:
> similar to what's discussed in [#20355 (comment)](https://github.com/neovim/neovim/pull/20355#discussion_r980511837) , can the root cause be fixed instead of sorting at the "leaf logic" here?

TBH I find it hard to say whether changing the structure of `auevents.nvim_specific` to make sure the output is deterministic is "fixing the root cause" or "leaking a presentation concern into the data layer" - but happy to adapt :)

> Similar question for LUA_GEN_PRG

Here I completely agree we should fix the 'root cause', which is the fact that the mpack'ed data contains string-keyed tables which are not serialized deterministically.

Reply:
> hard to say whether changing the structure of auevents.nvim_specific to make sure the output is deterministic is "fixing the root cause" or "leaking a presentation concern into the data layer" 

fair point, but if making the source structure deterministic has no downsides, better to do it there. Reduces entropy in general :)

Reply:
> but if making the source structure deterministic has no downsides

Looking at the latest of this PR, I see a downside: having to structure e.g. `keysets.lua` as a list instead of a dict is awkward. So I agree we need something like a `sorted()` function that iterates dicts in order.